electrum/gui/kivy/main_window.py

```diff
@@ -649,6 +649,10 @@ def load_wallet_by_name(self, path):
             return
         if self.wallet and self.wallet.storage.path == path:
             return
+        if self.password and self.electrum_config.get('use_single_password'):
+            storage = WalletStorage(path)
+            storage.check_password(self.password)
```

Comment:
what if wallet is only keystore-encrypted?

Reply:
This case is handled in `check_password_for_directory`. `self._use_single_password` is set only if the password was checked against all wallets and keystores. Here check_password is called for its side effect, to decrypt the storage. I will add a comment to clarify.